### ISSUE TITLE
Test your code

### ISSUE THREAD

**Issue by edouard-lopez** [bpkg/bpkg#72]
Please test your code using a bash framework such as [bash_unit](https://github.com/pgrange/bash_unit) or [any another you like](https://github.com/awesome-lists/awesome-bash#shell-script-development).

This will improve the **code quality** and **prevent regression**.

**Comment by joseluis** [comment]
I agree with @edouard-lopez that adding unit tests is important long-term. I personally like [Bats](https://github.com/sstephenson/bats) and [assert.sh](https://github.com/lehmannro/assert.sh) but any would do the job. Also it would be nice to connect the repo with [travis-ci](https://travis-ci.org) for automatic testing.

Also it would be good to review the code quality with [shellcheck](https://www.shellcheck.net/), and fix the few issues present. Also you could to connect the repo with [code-climate](https://codeclimate.com) for automating review.

**Comment by Potherca** [comment]
Just opened #112 to add tests to `getdeps`

**Comment by Potherca** [comment]
Started work for `utils` in #113

**Comment by Potherca** [comment]
Note to self: We should check which files caused the most issues lately and prioritize those for testing first!

**Comment by Potherca** [comment]
Well, running this:
```sh
git log --oneline --stat | grep '|' | cut -d ' ' -f2 | sort | uniq -c | sort -r -h
```
The top 5 bash files are:

1. 65 `lib/install/install.sh`
2. 40 `bpkg.sh`
3. 21 `lib/init/init.sh`

 Limited to the last 50 commit, that is:
 
 1. 22 `lib/install/install.sh`
 2. 12 `lib/init/init.sh`
 3. 10 `lib/show/show.sh`
 
So I think `lib/install/install.sh`, `bpkg.sh`, `lib/init/init.sh` and `lib/show/show.sh` will have to come first...
